### PR TITLE
[NADE] Enable Code Generation by default in Snow App Bundle

### PR DESCRIPTION
**Pull request by sfc-gh-bgoel**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Snowpark Annotation Processing in Native Apps through Code Generation of SQL is hidden behind a feature flag up until 2.4.0. But with 2.5.0, we want to make this widely available, without any feature flags. This PR removes the condition, and also adds an E2E test that ensures we get to a working app state.

### CODE REVIEW

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def process(
 
                 with open(sql_file, "a") as file:
                     file.write(
-                        f"-- Generated by the Snowflake CLI from {relative_py_file}\n"
+                        f"\n-- Generated by the Snowflake CLI from {relative_py_file}\n"
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py:227]: but why? This just adds a blank line at the top of every SQL file we generate. It just looks odd.

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def _execute_in_sandbox(
 
     if completed_process.returncode != 0:
         cc.warning(
-            f"Could not fetch Snowpark objects from {py_file} due to the following error:\n {completed_process.stderr}"
+            f"Could not fetch Snowpark objects from {py_file} due to the following Snowpark-internal error:\n {completed_process.stderr}"
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py:143]: errors here aren't necessarily snowpark

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja [hunk 3]
```diff
@@ -141,7 +141,8 @@ def __snowflake_internal_create_extension_fn_registration_callback():
         collected_extension_fn_json_list, extension_function_properties
     ):
         extension_fn_json = __snowflake_internal_extension_fn_to_json(extension_function_properties)
-        collected_extension_fn_json_list.append(extension_fn_json)
+        if extension_fn_json: # Do not append if extension_fn_json is None
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja:144]: when does that happen?

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja [hunk 4]
```diff
@@ -101,7 +101,7 @@ def __snowflake_internal_create_extension_fn_registration_callback():
 
 
     def __snowflake_internal_extension_fn_to_json(extension_fn):
-        if not isinstance(extension_fn.func, Callable):
+        if not (isinstance(extension_fn.func, Callable) or isinstance(extension_fn.func, Tuple)):
             # Unsupported case: extension function is a tuple
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja:105]: nit: this seems outdated, please update too

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py [hunk 5]
```diff
@@ -461,15 +459,20 @@ def generate_grant_sql_ddl_statements(
...
+            f"Skipping generation of 'GRANT USAGE ON ...' SQL statement for {extension_fn.function_type.upper()} {extension_fn.handler} due to lack of application roles."
         )
         return None
 
     grant_sql_statements = []
+    object_type = (
+        "PROCEDURE"
+        if extension_fn.function_type == ExtensionFunctionTypeEnum.PROCEDURE
+        else "FUNCTION"
+    )
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py:471]: what was wrong with making the call? I don't see that utility getting removed, so I'm a bit confused.

Comment (sfc-gh-bgoel) [reply src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py:471]: Apparently, while you do `CREATE OR REPLACE AGGREGATE FUNCTION` for create DDL, grant DDL needs to be `GRANT USAGE ON FUNCTION TO ...`, which is why we do not use the utility for grants.

File: src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py (hunk 2 above)
Comment (sfc-gh-bdufour) [reply src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py]: Errors that can go to stderr include bugs in our driver and just Python errors in user code. Labelling them all as Snowpark would be misleading.

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -21,6 +21,7 @@
 
 ## New additions
 * Added `snow app bundle` command that prepares a local folder in the project directory with artifacts to be uploaded to a stage as part of creating a Snowflake Native App.
+  * Added a new Snowpark Annotation Processing feature for a user to annotate their Python code with Snowpark Python annotations, which are used by the Snowflake CLI to generate corresponding SQL to create extension functions and append them to their setup script file for a Snowflake Native App.
```
Comment (sfc-gh-bdufour) [inline RELEASE-NOTES.md:24]: Seems a bit low level. How about:

```
Snowflake Native App projects can now optionally generate CREATE FUNCTION or CREATE PROCEDURE declarations in setup scripts from Snowpark python code that includes decorators (e.g. @sproc, @udf).
```

**Review by sfc-gh-bdufour — APPROVED** [review]

**Review by sfc-gh-bdufour — APPROVED** [review]

**Pull request merged**